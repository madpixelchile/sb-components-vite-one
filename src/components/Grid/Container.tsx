import { StyledGrid } from "./StyledGrid";

export interface ContainerProps{
    children?: React.ReactNode;
    className?: string;
    [key: string]: any;
}

export const Container = ( { children = null, className = '', ...props } : ContainerProps) => {
  return (
    <>
        {  <StyledGrid className={`container ${className}`} {...props} >{ children }</StyledGrid>  }
    </>
  )
}

export default Container;

