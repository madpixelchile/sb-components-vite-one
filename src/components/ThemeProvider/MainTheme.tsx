

import { ThemeProvider } from 'styled-components';
import { StyledThemeProvider } from './StyledThemeProvider';
import { darkMode, lightMode } from '../../styles/theme';

export interface MainThemeProps {
  children?: React.ReactNode;
  theme?: string | {},
  [key: string]: any;
}

export const MainTheme = ({ children = null, theme = 'light' }:MainThemeProps) => {

  let selectedTheme: string | object = lightMode;

  if(typeof theme !== 'object'){
    if(theme === 'light'){
      //No object - string light - default lightModeAdded
      selectedTheme = lightMode;
    }else if(theme === 'dark'){
      //No object - string dark -  default darkModeAdded
      selectedTheme = darkMode;
    }
  }else if(typeof theme === 'object'){
    if(Object.keys(theme).length === 0){ //!PENDING Compare with the final BASE required properties
      //Empty object - Default lightMode added
      selectedTheme = lightMode;
    }else{
      //Full object - CustomMode or theme added
      selectedTheme = theme;
    }
  }

  return (
    <>
        <ThemeProvider theme={ selectedTheme }>
            <StyledThemeProvider/>
            { children }
        </ThemeProvider>
    </>
  )
}

export default  MainTheme;
