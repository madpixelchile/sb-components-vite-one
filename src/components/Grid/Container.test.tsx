
import { screen, render } from '@testing-library/react'; //screen
import { Container } from './Container';
import MainTheme from '../ThemeProvider/MainTheme';

describe('Testing Containers', () => {

  it('Testing props', () => {

    const { getByText, container } = render(<MainTheme><Container className='container-fluid'>Children</Container></MainTheme>);
    const childrenElement = getByText('Children');
    expect(childrenElement.innerHTML).toBe('Children');

    const classProp = container.getElementsByClassName('container-fluid');
    expect(classProp.length).toBe(1)

    expect(container).toMatchSnapshot();

  })

  it('Testing classname container', () => {

    const { container } = render(<MainTheme><Container>Children</Container></MainTheme>);

    const classProp = container.getElementsByClassName('container');
    expect(classProp.length).toBe(1)

  })

  it('Testing extra props', () => {
    const { getByLabelText } = render(<MainTheme><Container aria-label={'extra-prop'}>Children</Container></MainTheme>);
    const ariaElement = getByLabelText('extra-prop');
    expect(ariaElement).toBeTruthy()
    screen.debug()
  })


})