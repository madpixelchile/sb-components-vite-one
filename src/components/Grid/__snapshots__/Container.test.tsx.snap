// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Containers Testing props 1`] = `
<div>
  <div
    class="sc-bdnyFh ktBSwJ container-fluid "
  >
    Children
  </div>
</div>
`;
